intel_extension_for_transformers/llm/runtime/graph/__init__.py

```diff
@@ -109,34 +126,36 @@ def generate(self, input_ids, streamer=None, interactive=False, ignore_prompt=Fa
             ret = input_ids.tolist()
 
         beam_search = False
-        if ("num_beams" in kwargs and kwargs["num_beams"] > 1) and not \
-            kwargs.get("do_sample", False):
+        if ("num_beams" in generate_kwargs and generate_kwargs["num_beams"] > 1) and not \
+            generate_kwargs.get("do_sample", False):
             beam_search = True
         if not beam_search:
             # TODO support multi batch
             assert input_ids.shape[0] == 1, "Unsupport multi-batch input ids."
+        
         if streamer:
-            if beam_search:
-                print("ERROR, can not use streamer when use beam search for generation!")
-                import sys
-                sys.exit(1)
+            assert input_ids.shape[0] == 1, "Streamer only supports batch size 1."
```

Comment:
So we can use `streamer` when doing beam search?

Reply:
cannot, update f71da91bf8c78a6d7376675d10d67a1996c10495